### DISCUSSION TITLE
Still seeing Ads on stream or video player is paused until ads has been played

### DISCUSSION THREAD

**Discussion by YaliHorde** [streamlink/streamlink-twitch-gui#844] [General]
Ads are still being played or displayed when using MPCHC player.
 Isnt Streamlink updated to account for newest Twitch API?
 If this program doesnt work, why wouldnt I just use Twitch website instead..
 
 Tried using Python and Streamlink script, Passthrough and other options. Nothing helps in avoiding ads.
 Andone have a clue what to do here?

**Comment by bastimeyer** [comment]
Wrong repo. Also not a particularly nice way of asking and demanding stuff on a free software project. Let me answer anyway...

**Streamlink Twitch GUI** is just a browser for Twitch and launcher for **Streamlink**. It has nothing to do with streaming. This is done by Streamlink and it's described pretty much everywhere here.

There are notes regarding preventing and filtering ads on Twitch via Streamlink in Streamlink Twitch GUI's [wiki](https://github.com/streamlink/streamlink-twitch-gui/wiki/Embedded-ads) and there's [a pinned thread](https://github.com/streamlink/streamlink-twitch-gui/issues/754) on its issue tracker.

As you can see on the Streamlink issue tracker (which is linked on both pages), there was a recent change on Twitch which already got fixed. So if you can't wait until the next release of Streamlink, then you have to install the latest version of Streamlink from its master branch. This is fully explained in the Streamlink documentation. Downgrading Streamlink to 2.3.0 could also be an option if I'm not mistaken.

Relevant threads on the Streamlink issue tracker:
- https://github.com/streamlink/streamlink/issues/4106#issuecomment-974797278
- https://github.com/streamlink/streamlink/pull/4194

Twitch embedded ads meta thread on Streamlink's issue tracker:
https://github.com/streamlink/streamlink/issues/3210

Streamlink install documentation:
https://streamlink.github.io/install.html
